yb-voyager/src/monitor/monitor_target_health.go

```diff
@@ -0,0 +1,203 @@
+		//which weren't present the starting and during the run it came back.
+		if !slices.Contains(lo.Keys(nodeStatuses), conf.Host) {
+			nodeStatuses[conf.Host] = true
+		}
+	}
+	if len(downNodes) > 0 {
+		downNodeMsg := fmt.Sprintf("Following nodes are not healthy, please check them - [%v]", strings.Join(downNodes, ", "))
+		err := yb.RemoveConnectionsForHosts(downNodes)
+		if err != nil {
+			return downNodeMsg, fmt.Errorf("error while removing the connections for the down nodes[%v]: %v", downNodes, err)
+		}
+		return downNodeMsg, nil
+	}
+	return "", nil
+}
+
+/*
+If disk space on any node is less than 10%, error out the import and let the user increase the disk
+
+//TODO: enhance yb_server_metrics() to give these metrics.
+*/
+func monitorDiskUsageAndAbort(yb TargetDBForMonitorHealth, skip utils.BoolStr) error {
+	if skip {
+		return nil
+	}
+	nodeMetrics, err := yb.GetClusterMetrics()
+	if err != nil {
+		return fmt.Errorf("error fetching cluster metrics: %v", err)
+	}
+	var totalFreeDisk, totalDisk int64
+	for _, metrics := range nodeMetrics {
+		if metrics.Status != "OK" {
+			continue
+		}
+		nodeFreeDisk, err := strconv.ParseInt(metrics.Metrics[METRIC_FREE_DISK_SPACE], 10, 64)
+		if err != nil {
+			return fmt.Errorf("parsing free disk space metric as int: %w", err)
+		}
+		nodeTotalDisk, err := strconv.ParseInt(metrics.Metrics[METRIC_TOTAL_DISK_SPACE], 10, 64)
+		if err != nil {
+			return fmt.Errorf("parsing total disk space metric as int: %w", err)
+		}
+		totalFreeDisk += nodeFreeDisk
+		totalDisk += nodeTotalDisk
+		freeDiskPct := (nodeFreeDisk * 100) / nodeTotalDisk
+		if freeDiskPct < FREE_DISK_SPACE_THREASHOLD {
+			utils.ErrExit("Free disk space available on the target cluster is low, increase the disk space on the target cluster and re-run the import.")
+		}
+	}
+	return nil
+}
+
+/*
+If any of the replication streams are enabled on the cluster during the import -
+let the user know to enable it after import data is done to avoid wal or disk space issues because of replication
+
+Caveats -
+1. For all deployments - only logical replication info is easily detectable - using the pg_replciation_slots for getting num of logical replication slots
+2. For YBA/yugabyted - with above also trying to figure the xcluster replication streams/CDC gRPC streams using the yb-client wrapper jar to fetch num of all cdc streams on the cluster
+*/
+func monitorReplicationOnTarget(yb TargetDBForMonitorHealth, skip utils.BoolStr) error {
+	if skip {
+		return nil
+	}
+	numOfSlots, err := yb.NumOfLogicalReplicationSlots()
+	if err != nil {
+		return fmt.Errorf("error fetching logical replication slots for checking if replication enabled - %s", err)
+	}
+	ybServers := lo.Keys(nodeStatuses)
+	addresses := strings.Join(ybServers, ":7100,")
+	// no need to specify the empty params as we need to only get num of cdc streams
+	ybClient := dbzm.NewYugabyteDBCDCClient("", addresses, "", "", "", nil)
```

Comment:
Doesn't make sense to initialize this every single time. (even though I suspect it might be lightweight). 

Can we initialize this once in the caller function, and pass it down to this function? 
Or use a global variable and check that it is inited. 

Alternatively, one option could be to create a struct for monitoring?
We seem to be saving node statuses. we need a handle to the ybclient to be able to make that call to get CDC streams..